Source/player.cpp

```diff
@@ -3233,36 +3233,33 @@ void __fastcall ArmorDur(int pnum)
 	}
 }
 
-int __fastcall PM_DoDeath(int pnum)
+BOOL __fastcall PM_DoDeath(int pnum)
 {
-	int v1; // edi
-	int v2; // esi
-	int v3; // ecx
-	int v4; // eax
-	int v5; // eax
-
-	v1 = pnum;
-	if ( (unsigned int)pnum >= MAX_PLRS )
+	if ( (DWORD)pnum >= MAX_PLRS ) {
 		TermMsg("PM_DoDeath: illegal player %d", pnum);
-	v2 = v1;
-	if ( plr[v1]._pVar8 >= 2 * plr[v1]._pDFrames )
-	{
-		if ( deathdelay > 1 && v1 == myplr && --deathdelay == 1 )
-		{
-			deathflag = 1;
-			if ( gbMaxPlayers == 1 )
-				gamemenu_previous();
+	}
+
+	if ( plr[pnum]._pVar8 >= 2 * plr[pnum]._pDFrames ) {
+		if ( deathdelay > 1 && pnum == myplr ) {
+			deathdelay--;
+			if ( deathdelay == 1 ) {
+				deathflag = 1;
+				if ( gbMaxPlayers == 1 ) {
+					gamemenu_previous();
+				}
+			}
 		}
-		v3 = plr[v2].WorldY;
-		plr[v2]._pAnimFrame = plr[v2]._pAnimLen;
-		v4 = plr[v2].WorldX;
-		plr[v2]._pAnimDelay = 10000;
-		dFlags[v4][v3] |= 4u;
-	}
-	v5 = plr[v2]._pVar8;
-	if ( v5 < 100 )
-		plr[v2]._pVar8 = v5 + 1;
-	return 0;
+
+		plr[pnum]._pAnimFrame = plr[pnum]._pAnimLen;
+		plr[pnum]._pAnimDelay = 10000;
+		dFlags[plr[pnum].WorldX][plr[pnum].WorldY] |= 4;

```

Comment:
I think we should create an enum for the dungeon flag bitfield.

Here are some notes from @NiteKat.

```
0x01 // unknown.
0x02 // Monster AI uses this flag. If set, any monster on that tile will become active if it was not. Gets set within a radius of the player, so that nearby monsters activate. Not all AI types use this flag - Hidden type enemies do not, for example, and will activate even if this flag is at 0 for their tile (i suspect the enemies that do not use this flag are just always active)
0x04 // Indicates a player corpse is there.
0x08 // Uknown. Seems to get set in some spots by the game by default.
0x10 // Unknown.
0x20 //Unkown.
0x40 // Denotes if your character has vision of that tile or not.
0x80 // Related to auto map. Forcing the bit set stops the Automap from revealing new areas as you explore, so maybe this marks if you've checked an area or not? Forcing it to 0 all the time seems to have no impact to the game, Automap generates as normal. Probably just causes extra calls to some function as you explore.
```

And here are some notes from @7i and myself.

```c
typedef enum {
	DFLAG_01         = 0x01 // NOT SAVED. Used by missiles.
	DFLAG_02         = 0x02 // NOT SAVED. Potentially Line of Sight Used in AI code. Tracking active monsters. Lighting clears all except DFLAG_02 and DFLAG_40
	DFLAG_04         = 0x04 // NOT SAVED. Dead player?
	DFLAG_08         = 0x08 // SAVED. dFlag sum in debug. Dungeon generation? used in gendung.
	DFLAG_MONSTER_LR = 0x10 // SAVED.
	DFLAG_20         = 0x20 // SAVED.
	DFLAG_40         = 0x40 // SAVED. Lighting clears all except DFLAG_02 and DFLAG_40
	DFLAG_80         = 0x80 // SAVED. Automap something? Clears everything on init except DFLAG_80
} dflag;
```

These can be used as a basis.

Reply:
Perhaps we should do this when more code has been cleaned up and it's more clear what the individual flags do?

Reply:
0x20 is unset during PM_DoWalk3 and RemovePlrFromMap

Reply:
> Perhaps we should do this when more code has been cleaned up and it's more clear what the individual flags do?

That works too.

For now we could simply name flags, e.g.

```c
typedef enum {
   DFLAG_20 = 0x20,
} dflag;

And later rename the flags when we know what `DFLAG_20` does. The rationale being that it's easier to search and replace `DFLAG_20`, then `0x20`.

Reply:
Went threw all the places that use dFlags, hers and over view of what I found:
DFLAG_01 is set on MissileCreate, and removed on ClearMissile
DFLAG_02 is set during DoVision and cleared during DoUnVision
DFLAG_04 is used by DrawDeadPlayer
DFLAG_08 is used in everything related to dungeon prefabs, like ThemeRooms, MiniSets, Monsters, Items etc.
DFLAG_10 is used for monsteres and towneres
DFLAG_20 is updated each time the player moves, even has a FixPlrWalkTags for clearning it
DFLAG_40 updated by DoVision and DoUnVision, checked by CheckCursMove, DrawDeadPlayer, DrawMonster
DFLAG_80 cleared by InitAutomap set by DoVision if doautomap=true

Based on that I have come up with the following names:
```
enum dflag
{
	DFLAG_MISSILE     = 0x01
	DFLAG_VISIBLE     = 0x02
	DFLAG_DEAD_PLAYER = 0x04
	DFLAG_POPULATED   = 0x08
	DFLAG_MONSTER     = 0x10
	DFLAG_PLAYER      = 0x20
	DFLAG_LIT         = 0x40
	DFLAG_SEEN        = 0x80
}
```

Reply:
There are also checks for 0x40 next to plr[myplr]._pInfraFlag

Reply:
> DFLAG_SEEN

Bike shedding, but if used for the automap, how about `DFLAG_EXPLORED`?

Reply:
Work on 30 line function: diff +164/-156 :D

DFLAG_SEEN as now been renamed to DFLAG_EXPLORED any thing else? :)